.travis.yml

```diff
@@ -30,7 +31,6 @@ before_install:
 
 install:
 - cabal new-update -v
-- cabal new-configure --enable-tests --disable-optimization --write-ghc-environment-files=always --jobs=2
 - cabal new-build --only-dependencies
```

Comment:
Likewise, I don’t understand this part either.

1. Is this actually building the tests’ dependencies at this point, or do they get built when we do the `new-run` calls later on?

2. _Why_ does `configure` result in so much being rebuilt? Where are those extra 20min of builds coming from?

3. Would we still see unnecessary rebuilds if we kept the configure step here, but added `cabal.project.local` to the cache?

4. What if we added the `--enable-tests` &c. flags to this line instead of using `configure`?

5. Can we move `--write-ghc-environment-files=always` & `--jobs=2` into the global cabal configuration file? (And then cache that as well.) Does that help?

Reply:
I note that haskell-ci doesn’t run configure; instead, it passes flags directly to the build command(s):

https://github.com/haskell-CI/haskell-ci/blob/master/.travis.yml#L148-L149

Reply:
1. The presence of `tests: True` in `cabal.project.local` means that tests’ dependencies.
2. We don’t cache the `cabal.project.local` file. Thus, when we call `cabal configure`, it creates a new `cabal.project.local` file, and invalidates the whole cache. This causes all the pinned and local dependencies to rebuild (the dreaded `proto3-suite-0.2.0 (lib) (configuration changed)` entries. My local testing shows that calling `cabal build; cabal configure --jobs=8 --disable-optimization; cabal build` didn’t cause a rebuild, so I _think_ this is a result of `cabal` having to generate a new `cabal.project.local` file.
3. I don’t think so (see above)
4. We could do that.
5. I think moving those into the global cabal file would be a net negative (we don’t know anything about the kinds of systems `semantic` might be running on, why specify any `jobs` settings?)